include/aws/event-stream/event_stream_rpc_client.h

```diff
@@ -0,0 +1,181 @@
+AWS_EVENT_STREAM_API void aws_event_stream_rpc_client_connection_acquire(
+    const struct aws_event_stream_rpc_client_connection *connection);
+AWS_EVENT_STREAM_API void aws_event_stream_rpc_client_connection_release(
+    const struct aws_event_stream_rpc_client_connection *connection);
+
+/**
+ * Closes the connection if it is open and aws_event_stream_rpc_client_connection_options::on_connection_shutdown will
+ * be invoked upon shutdown. shutdown_error_code will indicate the reason for shutdown. For a graceful shutdown pass 0
+ * or AWS_ERROR_SUCCESS.
+ */
+AWS_EVENT_STREAM_API void aws_event_stream_rpc_client_connection_close(
+    struct aws_event_stream_rpc_client_connection *connection,
+    int shutdown_error_code);
+
+/**
+ * Returns true if the connection is already closed.
+ */
+AWS_EVENT_STREAM_API bool aws_event_stream_rpc_client_connection_is_closed(
+    const struct aws_event_stream_rpc_client_connection *connection);
+
+/**
+ * Sends a message on the connection. These must be connection level messages (not application messages).
+ *
+ * flush_fn will be invoked when the message has been successfully writen to the wire or when it fails.
+ *
+ * returns AWS_OP_SUCCESS if the message was successfully created and queued, and in that case flush_fn will always be
+ * invoked. Otherwise, flush_fn will not be invoked.
+ */
+AWS_EVENT_STREAM_API int aws_event_stream_rpc_client_connection_send_protocol_message(
+    struct aws_event_stream_rpc_client_connection *connection,
+    const struct aws_event_stream_rpc_message_args *message_args,
+    aws_event_stream_rpc_client_message_flush_fn *flush_fn,
+    void *user_data);
+
+/**
+ * Create a new stream. continuation_option's callbacks will not be invoked, and nothing will be sent across the wire
+ * until aws_event_stream_rpc_client_continuation_activate() is invoked.
+ *
+ * returns an instance of a aws_event_stream_rpc_client_continuation_token on success with a reference count of 1. You
+ * must call aws_event_stream_rpc_client_continuation_release() when you're finished with it. Returns NULL on failure.
+ */
+AWS_EVENT_STREAM_API struct aws_event_stream_rpc_client_continuation_token *
+    aws_event_stream_rpc_client_connection_new_stream(
+        struct aws_event_stream_rpc_client_connection *connection,
+        const struct aws_event_stream_rpc_client_stream_continuation_options *continuation_options);
+AWS_EVENT_STREAM_API void aws_event_stream_rpc_client_continuation_acquire(
+    const struct aws_event_stream_rpc_client_continuation_token *continuation);
+AWS_EVENT_STREAM_API void aws_event_stream_rpc_client_continuation_release(
+    const struct aws_event_stream_rpc_client_continuation_token *continuation);
+
+/**
+ * returns true if the continuation has been closed.
+ */
+AWS_EVENT_STREAM_API bool aws_event_stream_rpc_client_continuation_is_closed(
+    const struct aws_event_stream_rpc_client_continuation_token *continuation);
```

Comment:
Confusing: this returns true if it's closed **OR CLOSING**

The user might believe that is_closed() tells whether the connection is done shutting down (this is also what the comments say)

In HTTP we have `is_open()` instead, to avoid that confusion. It's true when the connection is good, and false when it's closing or closed

Reply:
continuation stayed as is, but connection moved to is_open